classes/ActionScheduler_AdminView.php

```diff
@@ -35,9 +35,42 @@ public function init() {
 			} else {
 				add_action( 'admin_menu', array( $self, 'register_menu' ) );
 			}
+			add_action( 'admin_notices', array( $self, 'check_past_due_actions' ) );
 		}
 	}
 
+	/**
+	 * Check if there are any past due action
+	 *
+	 * This method will check if there are past due. Because Action Scheduler is action agnostic,
+	 * it will execute an action with a list of the actions.
+	 *
+	 * The action owners should be listening to this action, and do something to warn the users
+	 * about how to fix their WP-Cron or to run their actions manually.
+	 *
+	 * This method is executed by the `admin_notices`, so it is safe to render the admin notification
+	 * from the `action_scheduler_past_due_action` action.
+	 *
+	 */
+	public function check_past_due_actions() {
+		$instance = ActionScheduler_Store::instance();
+		if ( $cache = get_transient( __METHOD__ ) ) {
+			$action_ids = $cache['action_ids'];
+		} else {
+			$action_ids = $instance->query_actions( array(
+				'date' => new Datetime( apply_filters( 'action_scheduler_past_due_time', '-5 days' ) ),
+				'status' => ActionScheduler_Store::STATUS_PENDING,
+			) );
+			set_transient( __METHOD__, compact( 'action_ids' ) );
+		}
+
+		$action_ids = array( 2151 );
+		$actions = array();
+		foreach ( $action_ids as $id ) {
+			$actions[ $id ] = $instance->fetch_action( $id );
+		}
+		do_action( 'action_scheduler_past_due_action', $actions );
```

Comment:
What happens if there are 100,000 actions past due, but nothing hooking to `action_scheduler_past_due_action`? This code would still instantiate 100,000 actions on every admin page load.

I think the code here needs to trigger the `action_scheduler_past_due_action` and to instantiate the actions should run _only_ when:

1. `! empty( $action_ids )`
1.  there are callbacks on `'action_scheduler_past_due_action'` to avoid wasting time if there aren't; and

Reply:
I think we can limit to 10 or so IDs. We should check if there are more, in the admin, while rendering the rows. Perhaps we add a different background for tasks with past due. This notice thing is just to give a quick hint of things not being executed for some reason, using as little resources as possible.

Reply:
The gotcha with that is if the first X number of actions is for a plugin that doesn't hook to `action_scheduler_past_due_action`. For example, if _Memberships_ has 1,000 actions in the queue before the first action for Subscriptions, and they are all past due, then Subscriptions notice will never be displayed (at least it won't if Subscriptions is hooking only to its own actions, which it should do).

To get around that, you could have some way to register a "past due" handler/monitor for certain hooks. Then when querying for the past due actions, you can specify the hook/s which have a registered past due handler/monitor only. There's not really much need to get the rest. The only use case I can think of is to allow for a generic notice for Action Scheduler itself, which could be done by a handler registering with a hook that evaluates to `false` if the handler/monitor registration is designed to allow that.

Reply:
I love that idea. I´ll make it happen.